File: Editorial/3546_Equal_Sum_Grid_Partition_I/Go/2_Matrix_Rotation_Row_wise_Enumeration.go
func canPartitionGrid(grid [][]int) bool {
	var total int64 = 0
	m, n := len(grid), len(grid[0])
	for i := 0; i < m; i++ {
		for j := 0; j < n; j++ {
			total += int64(grid[i][j])
		}
	}
	for k := 0; k < 2; k++ {
		var sum int64 = 0
		m, n = len(grid), len(grid[0])
		for i := 0; i < m-1; i++ {
			for j := 0; j < n; j++ {
				sum += int64(grid[i][j])
			}
			if sum*2 == total {
				return true
			}
		}
		grid = rotation(grid)
	}
	return false
}

func rotation(grid [][]int) [][]int {
	m, n := len(grid), len(grid[0])
	tmp := make([][]int, n)
	for i := range tmp {
		tmp[i] = make([]int, m)
	}
	for i := 0; i < m; i++ {
		for j := 0; j < n; j++ {
			tmp[j][m-1-i] = grid[i][j]
		}
	}
	return tmp
}

File: Editorial/3546_Equal_Sum_Grid_Partition_I/CSharp/2_Matrix_Rotation_Row_wise_Enumeration.cs
public class Solution {
    public bool CanPartitionGrid(int[][] grid) {
        long total = 0;
        int m = grid.Length;
        int n = grid[0].Length;
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                total += grid[i][j];
            }
        }
        for (int k = 0; k < 2; k++) {
            long sum = 0;
            m = grid.Length;
            n = grid[0].Length;
            for (int i = 0; i < m - 1; i++) {
                for (int j = 0; j < n; j++) {
                    sum += grid[i][j];
                }
                if (sum * 2 == total) {
                    return true;
                }
            }
            grid = Rotation(grid);
        }
        return false;
    }

    public int[][] Rotation(int[][] grid) {
        int m = grid.Length, n = grid[0].Length;
        int[][] tmp = new int [n][];
        for (int i = 0; i < n; i++) {
            tmp[i] = new int[m];
        }
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                tmp[j][m - 1 - i] = grid[i][j];
            }
        }
        return tmp;
    }
}
